src/pages/Trade/components/OrderBookCard/MyTrades.tsx

```diff
@@ -1,49 +1,106 @@
-	trades: getMyTrades(state),
-	isLoading: getIsLoadingMyTrades(state),
-	isLoaded: getIsLoadedMyTrades(state),
+	walletAddress: getCurrentWalletAddress(state),
+	isWalletConnected: getIsWalletConnected(state),
 });
 
-const mapDispatchToProps = {
-	fetchMyTradesRequest,
-};
-
-const connector = connect(mapStateToProps, mapDispatchToProps);
+const connector = connect(mapStateToProps);
 
 type PropsFromRedux = ConnectedProps<typeof connector>;
 
 type MyTradesProps = PropsFromRedux;
 
-const MyTrades: FC<MyTradesProps> = ({ fetchMyTradesRequest, trades, isLoading, isLoaded }) => {
-	useEffect(() => {
-		fetchMyTradesRequest();
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, []);
+type SettledTrade = {
+	exchangeTimestamp: number;
+	dest: CurrencyKey;
+	src: CurrencyKey;
+	amount: number;
+	rebate: number;
+	reclaim: number;
+};
+type SettledTrades = SettledTrade[];
+
+const MyTrades: FC<MyTradesProps> = ({ walletAddress, isWalletConnected }) => {
+	const settledTradesQuery = useQuery<SettledTrades, any>(
+		QUERY_KEYS.HistoricalTrades.MyTrades(walletAddress || ''),
+		() =>
+			snxData.exchanger.exchangeEntriesSettled({
+				from: walletAddress,
+				max: MAX_TRADES,
+			}),
+		{
+			refetchInterval: REFRESH_INTERVAL,
+		}
+	);
+	const tradesQuery = useQuery<HistoricalTrades, any>(
+		QUERY_KEYS.HistoricalTrades.MyTrades(walletAddress || ''),
+		() => {
+			const trades = snxData.exchanges.since({
+				fromAddress: walletAddress,
+				maxBlock: Number.MAX_SAFE_INTEGER,
+				max: MAX_TRADES,
+			});
+			for (let i = 0; i < trades.length; i++) {
+				if (!!settledTradesQuery.data) {
+					const settledTrade = settledTradesQuery.data.find(
+						(settledTrade) =>
+							trades[i].timestamp === settledTrade.exchangeTimestamp &&
+							settledTrade.dest === trades[i].toCurrencyKey &&
+							settledTrade.src === trades[i].fromCurrencyKey &&
+							trades[i].fromAmount === settledTrade.amount
+					);
 
-	useInterval(() => {
-		fetchMyTradesRequest();
-	}, REFRESH_INTERVAL);
+					trades[i].isSettled = false;
+					if (!!settledTrade) {
+						trades[i].rebate = settledTrade.rebate;
+						trades[i].reclaim = settledTrade.reclaim;
 
+						// special case for when the currency is priced in sUSD
+						const feeReclaimRebateAmount =
+							trades[i].toCurrencyKey === SYNTHS_MAP.sUSD
+								? settledTrade.rebate - settledTrade.reclaim
+								: settledTrade.reclaim - settledTrade.rebate;
+
+						// ( shiftAmount / amount ) * price -> gets us the price shift
+						// to get the new price, we just add the price shift (which might be a negative or positive number)
+						trades[i].settledPrice =
+							(feeReclaimRebateAmount / trades[i].toAmount) * trades[i].price + trades[i].price;
+						trades[i].isSettled = true;
+						trades[i].amount = Math.abs(feeReclaimRebateAmount);
+					}
+				}
+			}
+			return trades;
+		},
+		{
+			refetchInterval: REFRESH_INTERVAL,
+			enabled: !!settledTradesQuery.data,
```

Comment:
After `settledTradesQuery` executes once `enabled` will always be true which is why we need `refetchInterval` here as well correct?